runhouse/resources/hardware/cluster.py

```diff
@@ -994,13 +1033,23 @@ def run(
         if not run_name:
             # If not creating a Run then just run the commands via SSH and return
             return self._run_commands_with_ssh(
-                commands, cmd_prefix, stream_logs, port_forward, require_outputs
+                commands,
```

Comment:
nit: should `node` be first?

Reply:
made it optional since we can always default to using the head node's address, do you think that API should be more explicit?

Reply:
Optional sounds great to me, thanks